### PR TITLE
feat: Support asset v4 #FS-160

### PR DESCRIPTION
**Pull request by sb88k**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/FS-160" title="FS-160" target="_blank">StoryFS-160</a>  [iOS] As a user I like to share files in conversations with members from federated backends
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Support asset v4 endpoints for testing

### CODE REVIEW

File: Source/Assets/MockTransportSession+assets.swift [hunk 1]
```diff
@@ -28,4 +31,59 @@ extension MockTransportSession {
...
+    @objc(processAssetV4GetWithDomain:key:)
+    public func processAssetV4Get(with domain: String, key: String) -> ZMTransportResponse {
+        if let asset = MockAsset(in: managedObjectContext, forID: key, domain: domain) {
+            return ZMTransportResponse(imageData: asset.data, httpStatus: 200, transportSessionError: nil, headers: nil)
+        }
+        return ZMTransportResponse(payload: nil, httpStatus: 404, transportSessionError: nil)
+    }
+
+    @objc(processAssetV4DeleteWithDomain:key:)
+    public func processAssetV4Delete(with domain: String, key: String) -> ZMTransportResponse {
```
Comment (sb88k) [inline Source/Assets/MockTransportSession+assets.swift:81]: @typfel 
Was there a discussion about v4 delete? I couldn't find from [here](https://wearezeta.atlassian.net/wiki/spaces/CORE/pages/454525173/Federation+Assets+discussion+and+planning).

File: Source/Assets/MockTransportSession+assets.swift [hunk 2]
```diff
@@ -28,4 +31,59 @@ extension MockTransportSession {
...
+
+        if json["public"] as? Bool == false {
+            asset.token = UUID.create().transportString()
+        }
+
+        let payload = [
+            "key": asset.identifier,
+            "domain": domain,
+            "token": asset.token
+        ].compactMapValues { $0 } as ZMTransportData
```
Comment (sb88k) [inline Source/Assets/MockTransportSession+assets.swift:63]: I couldn't find the information about the token either for v4. Do we keep it as it is in v3?

File: Source/Assets/MockTransportSession+assets.swift [hunk 3]
```diff
@@ -28,4 +31,59 @@ extension MockTransportSession {
...
+
+        let location = String(format: "/asset/v4/%@", arguments: [asset.identifier])
+        return ZMTransportResponse(payload: payload,
+                                   httpStatus: 201,
+                                   transportSessionError: nil,
+                                   headers: ["Location": location])
```
Comment (sb88k) [inline Source/Assets/MockTransportSession+assets.swift:69]: Do we keep the location here?

File: Source/Assets/MockTransportSession+assets.swift (hunk 1 above)
Comment (sb88k) [reply Source/Assets/MockTransportSession+assets.swift:81]: Found answer from the backend ticket.
https://wearezeta.atlassian.net/jira/software/c/projects/FS/boards/119?modal=detail&selectedIssue=FS-306

File: Source/Assets/MockTransportSession+assets.swift (hunk 2 above)
Comment (typfel) [reply Source/Assets/MockTransportSession+assets.swift:63]: I'd assume we keep everything from v3 and that v4 is mostly extending v3 with the domain part.

File: Source/Assets/MockTransportSession+assets.swift (hunk 3 above)
Comment (typfel) [reply Source/Assets/MockTransportSession+assets.swift:69]: I'd assume we keep it.

**Review by typfel — APPROVED** [review]

**Review by johnxnguyen — APPROVED** [review]

**Comment by sb88k** [comment]
release minor

**Comment by zenkins** [comment]
Hello, @sb88k! Merging PR for command **release minor** for PR #**207**. Will create a **minor** release after merged.
Next version is 136.2.0.

**Pull request merged**